pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -69,7 +69,7 @@ func NewCodecForScheme(
 	encodeVersion []unversioned.GroupVersion,
 	decodeVersion []unversioned.GroupVersion,
 ) runtime.Codec {
-	return NewCodec(encoder, decoder, scheme, scheme, scheme, runtime.ObjectTyperToTyper(scheme), encodeVersion, decodeVersion)
+	return NewCodec(encoder, decoder, runtime.UnsafeObjectConvertor(scheme), scheme, scheme, runtime.ObjectTyperToTyper(scheme), encodeVersion, decodeVersion)
```

Comment:
If you are using a versioning codec, conversion is always "create
intermediate and then throw it away".  If you don't run conversion, then we
guarantee not to mutate your object, *unless* you give us an object that
mutates itself during serialization, in which case we hate you and you
deserve what you get.

On Fri, May 6, 2016 at 4:05 PM, David Eads <notifications@github.com> wrote:

> In pkg/runtime/serializer/versioning/versioning.go
> <https://github.com/kubernetes/kubernetes/pull/25018#discussion_r62383689>
> :
>
> > @@ -69,7 +69,7 @@ func NewCodecForScheme(
> >  	encodeVersion []unversioned.GroupVersion,
> >  	decodeVersion []unversioned.GroupVersion,
> >  ) runtime.Codec {
> > -	return NewCodec(encoder, decoder, scheme, scheme, scheme, runtime.ObjectTyperToTyper(scheme), encodeVersion, decodeVersion)
> > +	return NewCodec(encoder, decoder, runtime.UnsafeObjectConvertor(scheme), scheme, scheme, runtime.ObjectTyperToTyper(scheme), encodeVersion, decodeVersion)
>
> So are we assuming this safe by default since during a conversion step
> every sane thing I can think of throws away the original or is there some
> reason I'm not seeing that makes this always safe?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25018/files/ca669c5de0eba299d027ecb1c9d32155ef058038#r62383689>
>